### PR TITLE
GitLab validation 

### PR DESCRIPTION
**Pull request by lbarcziova**
Related to #401 

TODO:
- [x] update helm repo
- [x] create GL account
- [ ] add token + update variable in GitLab deployment repo

### CODE REVIEW

File: cron-jobs/packit-service-validation/packit-service-validation.py [hunk 1]
```diff
@@ -464,25 +695,15 @@ def check_comment(self):
...
-    if pr_for_push:
-        Testcase(
-            pr=pr_for_push[0], trigger=Trigger.push, deployment=deployment
-        ).run_test()
-
-    logging.info("Run testcase where the build is triggered by opening a new PR")
-    Testcase(deployment=deployment).run_test()
+    if getenv("GITLAB_TOKEN"):
+        logging.info("Running validation for GitLab.")
+        pass
```
Comment (jpopelka) [inline cron-jobs/packit-service-validation/packit-service-validation.py:700]: leftover or intentional?

File: cron-jobs/packit-service-validation/packit-service-validation.py [hunk 2]
```diff
@@ -375,79 +331,354 @@ def check_build(self, build_id):
...
+            commit_msg,
+            content,
+            contents.sha,
+            branch=branch,
+            committer=self.user,
+            author=self.user,
+        )
+
+
+class GitlabTestcase(Testcase):
```
Comment (lachmanfrantisek) [inline cron-jobs/packit-service-validation/packit-service-validation.py:556]: Oh, I didn't expected you need to reimplement all the functions..;(

Do you think we can implement something in OGR instead to not need to do all of the following differently?
(Most probably not possible with everything..)

File: cron-jobs/packit-service-validation/packit-service-validation.py [hunk 3]
```diff
@@ -375,79 +331,354 @@ def check_build(self, build_id):
...
+        self.test_case_kls(project=self.project, deployment=deployment).run_test()
+
+
+class GitlabTests(Tests):
+    test_case_kls = GitlabTestcase
+
+    def __init__(self):
+        gitlab_service = GitlabService(
+            token=getenv("GITLAB_TOKEN"), instance_url="https://gitlab.com"
+        )
```
Comment (lachmanfrantisek) [inline cron-jobs/packit-service-validation/packit-service-validation.py:668]:
```suggestion
    def __init__(self, instance_url="https://gitlab.com"):
        gitlab_service = GitlabService(
            token=getenv("GITLAB_TOKEN"), instance_url=instance_url
        )
```

Comment (lachmanfrantisek) [reply cron-jobs/packit-service-validation/packit-service-validation.py:668]: So we can support more instances in the future...

**Review by lachmanfrantisek — APPROVED** [review]
Thanks!

File: cron-jobs/packit-service-validation/packit-service-validation.py (hunk 1 above)
Comment (lbarcziova) [reply cron-jobs/packit-service-validation/packit-service-validation.py:700]: oh, thanks for spotting this, leftover

File: cron-jobs/packit-service-validation/packit-service-validation.py (hunk 2 above)
Comment (lbarcziova) [reply cron-jobs/packit-service-validation/packit-service-validation.py:556]: most of these are really relevant only for the script (e.g. determining whether status is pending/successful differs in Gitlab and Github, or creating an empty commit is done differently for them), but some methods could be implemented directly in ogr (e.g. deleting branch or creating a file)

**Pull request merged**